internal/controllers/topology/cluster/structuredmerge/options.go

```diff
@@ -60,3 +77,13 @@ type IgnorePaths []contract.Path
 func (i IgnorePaths) ApplyToHelper(opts *HelperOptions) {
 	opts.ignorePaths = i
 }
+
+// AllowedSpecPaths instruct the Helper to ignore everything in spec except given paths when computing a patch.
+// NOTE: ignorePaths are used to filter out fields nested inside allowedPaths, e.g.
+// spec.ControlPlaneEndpoint.
+type AllowedSpecPaths []contract.Path
+
+// ApplyToHelper applies this configuration to the given helper options.
+func (a AllowedSpecPaths) ApplyToHelper(opts *HelperOptions) {
+	opts.allowedPaths = append(allowedPathsMetadata, a...)
```

Comment:
adding `allowedPathsMetadata` when processing `AllowedSpecPaths` is kind of confusing. At least we should document it better, but this is kind of hard to follow with the defaulting implemented in serversidepathhelper.go